tonic/transforms.py

```diff
@@ -643,6 +643,35 @@ def __call__(self, events):
         )
 
 
+@dataclass()
+class ToBinaRep:
+    """Representation that takes T*B binary event frames to produce a sequence of T frames of N-bit numbers.
+    To do so, N binary frames are interpreted as a single frame of N-bit representation. Taken from the paper
+    Barchid et al. 2022, Bina-Rep Event Frames: a Simple and Effective Representation for Event-based cameras
+    https://arxiv.org/pdf/2202.13662.pdf
+    Parameters:
+        to_frame_transform (tonic.transforms.ToFrame): the ToFrame transform that creates the T*B binary event frames. Defaults to None.
+        n_frames (int): the number T of bina-rep frames.
+        n_bits (int): the number N of bits used in the N-bit representation.
+    """
+
+    n_frames: Optional[int] = 1
+    n_bits: Optional[int] = 8
+    to_frame_transform: Optional[ToFrame] = None
```

Comment:
it's good practice to re-use existing code, but I have the feeling that here the user is given to many options. Do you think the user will change ToFrame's overlap and include_incomplete options? Because if not, then I suggest adding `sensor_size` as a parameter and to call ToFrame internally, without exposing it to the user. So basically
```
transform = transforms.ToBinaRep(
            sensor_size=sensor_size,
            n_frames=n_frames,
            n_bits=n_bits,
        )
```
That will also take care of the __postinit__. It's not a problem that the dataclass is not frozen.

Reply:
The choice of allowing the user to choose the ToFrame parameters is mostly inspired from my experience (for the paper and for future works). I often finetune the parameters of `ToFrame()` depending on the DVS records I work with (e.g. the record of a table tennis match is a real pain when working with `n_time_bins`, as it creates event frames that can vary a lot in appearance when the ball is smashed or just sliced). That's why I felt that it was more usable to make the `ToFrame` processing completely exposed to fit the user's needs.

But as I am one of the author, I may not be aware of how wide the options are for a user that just wants to check how the transform works. If you are sure that closing `ToFrame` is a better option for user-friendliness, I can change it.

Reply:
I see, that makes sense. Let's keep it flexible! However, I'm still not a fan of how the frame transform is passed in a constructor. Reason for this is that I'd like to keep the API fairly consistent where possible. If a transform only works on a specific kind of data (raw events, frames,...), then normally the user has to chain the transforms. How about the following: 
* We call BinRep a 'transformation that works on frames', which only accepts as data numpy arrays of shape (C, H, W). 
* The transform chain would then look something like this:
```
import tonic.transforms as transforms

transform = transforms.Compose([
    transforms.ToFrame(
        sensor_size=sensor_size,
        n_time_bins=n_time_bins,
    ),
    transforms.ToBinaRep(
        n_frames=n_frames,
        n_bits=n_bits,
    ),
])
```
I feel like that would make it very clear to a user that BinaRep works on frames and reflects the general way of how to chain transforms in Tonic. How does that sound to you?

Reply:
Sounds good to me! I will update the code and documentation.

Reply:
Great, thank you. One last thing. To make it really easy for the user, could you also include an 'example' in the docstring much like here: https://github.com/neuromorphs/tonic/blob/develop/tonic/transforms.py#L14
You could take the example I posted above that includes the separate ToFrame transform. Thank you!

Reply:
Done.